### PR TITLE
Clarify terminology plugin/device in docs

### PR DESCRIPTION
**Pull request by mariaschuld**
This PR improves the docstrings to use PennyLane's latest way to define devices and plugins. Historically, plugins and devices were sometimes conflated, but now plugins, as well as PL core, *provide* one or multiple devices.

A thorough search for the term "plugins" only exposed two files with the outdated usage.  Especially the term "plugin device" is outdated, since devices can also be provided by core.

*Question: Shall we fix links and terminology in the changelog at all?*

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @mariaschuld! I agree with @thisac; I would leave the changelog as-is, as it reflects a historical log of the project.

Happy to merge this in when ready, I can see codecov is playing up

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -293,10 +293,10 @@ Isacsson, Josh Izaac, Nathan Killoran, Maria Schuld, Antal Száva, Nicola Vitucc
   ```
 
   See the `default.qubit.tf`
-  [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
+  [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.devices.DefaultQubitTF.html)
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:296]:
```suggestion
  [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -296,7 +296,7 @@ Isacsson, Josh Izaac, Nathan Killoran, Maria Schuld, Antal Száva, Nicola Vitucc
   [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
   for more details.
 
-* The [default.tensor plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
+* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/devices/default_tensor.py)
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:299]:
```suggestion
* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
```

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -296,7 +296,7 @@ Isacsson, Josh Izaac, Nathan Killoran, Maria Schuld, Antal Száva, Nicola Vitucc
   [documentation](https://pennylane.ai/en/stable/code/api/pennylane.beta.plugins.DefaultQubitTF.html)
   for more details.
 
-* The [default.tensor plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
+* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:299]:
```suggestion
* The [default.tensor device](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/devices/default_tensor.py)
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -1431,7 +1431,7 @@ Johannes Jakob Meyer, Maria Schuld, Antal Száva, Roeland Wiersema.
     which will alert the user that an operation with `num_wires = All`
     is being incorrectly.
 
-* The one-qubit rotations in `pennylane.plugins.default_qubit` no longer depend on Scipy's `expm`. Instead
+* The one-qubit rotations in `pennylane.devices.default_qubit` no longer depend on Scipy's `expm`. Instead
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:1434]:
```suggestion
* The one-qubit rotations in `pennylane.plugins.default_qubit` no longer depend on Scipy's `expm`. Instead
```

File: .github/CHANGELOG.md (hunk 3 above)
Comment (mariaschuld) [inline .github/CHANGELOG.md:299]:
```suggestion
* The [default.tensor plugin](https://github.com/XanaduAI/pennylane/blob/master/pennylane/beta/plugins/default_tensor.py)
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -1068,9 +1068,9 @@ Ville Bergholm, Josh Izaac, Johannes Jakob Meyer, Maria Schuld, Antal Száva.
   of an operator in the computational basis.
   [(#454)](https://github.com/XanaduAI/pennylane/pull/454)
 
-* Added a `QubitDevice` class which implements common functionalities of plugin devices such that
-  plugin devices can rely on these implementations. The new `QubitDevice` also includes
-  a new `execute` method, which allows for more convenient plugin design. In addition, `QubitDevice`
+* Added a `QubitDevice` class which implements common functionalities of devices such that
+  plugins can rely on these implementations. The new `QubitDevice` also includes
+  a new `execute` method, which allows for more convenient device design. In addition, `QubitDevice`
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:1073]:
```suggestion
* Added a `QubitDevice` class which implements common functionalities of plugin devices such that
  plugin devices can rely on these implementations. The new `QubitDevice` also includes
  a new `execute` method, which allows for more convenient plugin design. In addition, `QubitDevice`
```

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -1648,7 +1648,7 @@ Josh Izaac and Nathan Killoran.
 
 * Removed use of hardcoded values in the optimizers, made them parameters (see [#131](https://github.com/XanaduAI/pennylane/pull/131) and [#132](https://github.com/XanaduAI/pennylane/pull/132))
 * Created the new `PlaceholderExpectation`, to be used when both CV and qubit expval modules contain expectations with the same name
-* Provide the plugins a way to view the operation queue _before_ applying operations. This allows for on-the-fly modifications of
+* Provide a way for plugins to view the operation queue _before_ applying operations. This allows for on-the-fly modifications of
```
Comment (mariaschuld) [inline .github/CHANGELOG.md:1651]: Just fixing this error here

**Comment by mariaschuld** [comment]
@josh146 thanks, I reverted changes in the changelog...ready to merge!

**Pull request merged**